ongebo/java-coders-nairobi
# #156037248 UI/UX Revamp

### What does this PR do?
Improves the user interface and user experience
#### Description of the task to be completed?
- Add `grid_view_item.xml` for the views to be displayed in a grid layout
- Add Snackbar and CardView dependencies to `build.gradle`
- Display homepage profiles in a grid layout instead of linear layout
- Implement swipe to refresh feature in the main activity
- Display progress bar when data is being fetched from the API
- Display network errors using Snackbars instead of logging to the console
- Refactor: move activity files to the `view` package
#### How should this be manually tested?
- Run the app on this branch in Android Studio
- Observe the following updates:
  - The user profiles are displayed in a 2 column grid for portrait mode and a 3 column grid for
landscape mode
  - The homepage is refreshed when you swipe up
  - A progress indicator is displayed when data for the homepage is being fetched
  - An error is shown in case the data fetch is unsuccessful
#### Screenshots
##### GridView and swipe to refresh
![gridview + swipe to refresh](https://user-images.githubusercontent.com/34346070/54593058-b6fd6b80-4a3e-11e9-9ab7-542ea86d3b0c.gif)
##### Progress Bar
![ProgressBar](https://user-images.githubusercontent.com/34346070/54593082-c381c400-4a3e-11e9-813c-e3f6a9df76f5.gif)

#### What are the relevant Pivotal Tracker stories?
[#156037248](https://www.pivotaltracker.com/story/show/156037248)
